Authors: Oliver Seufer <debian@seufer.de>, gregor herrmann <gregoa@debian.org>
Description: deal with filenames with spaces
Bug: #488954

--- a/Rcs.pm
+++ b/Rcs.pm
@@ -526,7 +526,7 @@
         unless -e $rcsdiff_prog;
     return(_rcsError "rcsdiff program $rcsdiff_prog not executable")
         unless -x $rcsdiff_prog;
-    open(DIFF, "$rcsdiff_prog $param_str $arcfile $workfile |")
+    open(DIFF, "$rcsdiff_prog $param_str '$arcfile' '$workfile' |")
         or return(_rcsError "Can't fork $rcsdiff_prog: $!");
     my @diff_output = <DIFF>;
 
@@ -619,7 +619,7 @@
     my $archive_file = $rcsdir . $Dir_Sep . $arcfile;
     return(_rcsError "rlog program $rlogprog not found") unless -e $rlogprog;
     return(_rcsError "rlog program $rlogprog not executable") unless -x $rlogprog;
-    open(RLOG, "$rlogprog $param_str $archive_file |")
+    open(RLOG, "$rlogprog $param_str '$archive_file' |")
         or return(_rcsError "Can't fork $rlogprog: $!");
 
     my @logoutput = <RLOG>;
@@ -647,7 +647,7 @@
     my $archive_file = $rcsdir . $Dir_Sep . $arcfile;
     return(_rcsError "rcsmerge program $rcsmergeprog not found") unless -e $rcsmergeprog;
     return(_rcsError "rcsmerge program $rcsmergeprog not executable") unless -x $rcsmergeprog;
-    open(RCSMERGE, "$rcsmergeprog $param_str $archive_file |")
+    open(RCSMERGE, "$rcsmergeprog $param_str '$archive_file' |")
         or return(_rcsError "Can't fork $rcsmergeprog $!");
 
     my @logoutput = <RCSMERGE>;
